### ISSUE TITLE
Support JSON/Objects in Escaped Template Parts

### ISSUE THREAD

**Issue by ryansolid** [cristianbote/goober#136]
The scenario I'm talking about is:

```js
const Title = styled("h1")`
  font-weight: bold;
  ${p => ({ color: "dodgerblue" })};
`;
```
As far as I can tell this just gets serialized to `[Object object]`. This was brought to my attention due to someone trying to use Styled System(https://github.com/styled-system/styled-system) with Goober which works this way apparently. It lets developers replace common transformations with premade functions. Cool in concept at least.

**Comment by praneybehl** [comment]
Just curious, if this feature is planned?

**Comment by cristianbote** [comment]
Hey @praneybehl, thanks for your comment! I still haven't had a chance to look more deeply into it. If anyone wants to look into it, feel free! I would much appreciate it!

**Comment by cristianbote** [comment]
Actually @praneybehl took a quick look at the issue, and I think we could make `styled-system` work with goober as is. Check out this example https://codesandbox.io/s/goober-with-styled-system-mp0ex

Basically, instead of having to use tagged templates, we could use the object syntax and spread the results from the styled system functions. Pasting here the above example:
```jsx
import { h, render } from "preact";
import { color } from "styled-system";
import { styled, setup } from "goober";

const theme = {
  space: [0, 4, 8, 12, 16],
  colors: {
    red: "cyan",
    blue: "#336699"
  }
};

// Setup goober
setup(
  h, // pragma / createElement
  null, // autoprefixer
  () => theme // theme function
);

const Button = styled("button")((p) => ({
  border: 0,
  ...color(p)
}));

function App() {
  return (
    <div>
      <Button color="red" backgroundColor="blue">
        cyan button
      </Button>
    </div>
  );
}
```

Would love to learn your thoughts on this one.

**Comment by praneybehl** [comment]
Thanks @cristianbote, I guess it can be used as a workaround.
Maybe, I'll have a look to see what is needed to get it working with tagged templates.

Cheers 👍🏼

**Comment by cristianbote** [comment]
Alrighty! Due to popular demand and low footprint in size, this is working for tagged templates as well. Available in `2.0.9`. Thanks for reaching out and pursue this! Much appreciate it!
